enclave-runtime/src/test/sidechain_aura_tests.rs

```diff
@@ -0,0 +1,239 @@
+
+	info!("Create trusted operations..");
+	let sender = endowed_account();
+	let sender_with_low_balance = second_endowed_account();
+	let receiver = unendowed_account();
+	let transfered_amount: Balance = 1000;
+	let trusted_operation = encrypted_trusted_operation_transfer_balance(
+		ocall_api.as_ref(),
+		&shard_id,
+		&shielding_key,
+		sender,
+		receiver.public().into(),
+		transfered_amount,
+	);
+	let invalid_trusted_operation = encrypted_trusted_operation_transfer_balance(
+		ocall_api.as_ref(),
+		&shard_id,
+		&shielding_key,
+		sender_with_low_balance,
+		receiver.public().into(),
+		200000,
+	);
+	info!("Add trusted operations to TOP pool..");
+	let author_submit_future = async { rpc_author.submit_top(trusted_operation, shard_id).await };
+	executor::block_on(author_submit_future).unwrap();
+	let author_submit_future =
+		async { rpc_author.submit_top(invalid_trusted_operation, shard_id).await };
+	executor::block_on(author_submit_future).unwrap();
+
+	// Ensure we have exactly two trusted calls in our TOP pool, and no getters.
+	assert_eq!(2, rpc_author.get_pending_tops_separated(shard_id).unwrap().0.len());
+	assert!(rpc_author.get_pending_tops_separated(shard_id).unwrap().1.is_empty());
+
+	info!("Setup AURA SlotInfo");
+	let parentchain_header = ParentchainHeaderBuilder::default().build();
+	let timestamp = duration_now();
+	let slot = slot_from_time_stamp_and_duration(duration_now(), SLOT_DURATION);
+	let slot_info = SlotInfo::new(slot, timestamp, SLOT_DURATION, parentchain_header.clone());
+
+	info!("Test setup is done.");
+
+	let state_hash_before_block_production = get_state_hash(state_handler.as_ref(), &shard_id);
+
+	info!("Executing AURA on slot..");
+	let (blocks, opaque_calls) =
+		exec_aura_on_slot::<_, ParentchainBlock, SignedSidechainBlock, _, _>(
+			slot_info,
+			signer,
+			ocall_api.as_ref().clone(),
+			proposer_environment,
+			shards,
+		)
+		.unwrap();
+
+	assert_eq!(1, blocks.len());
+	assert_eq!(
+		state_hash_before_block_production,
+		get_state_hash(state_handler.as_ref(), &shard_id)
+	);
+
+	// Ensure that invalid calls are removed from pool. Valid calls should only be removed upon block import.
+	assert_eq!(1, rpc_author.get_pending_tops_separated(shard_id).unwrap().0.len());
+
+	info!("Executed AURA successfully. Sending blocks and extrinsics..");
+	let propose_to_block_import_ocall_api =
+		ProposeToImportOCallApi::new(parentchain_header, block_importer);
+
+	send_blocks_and_extrinsics::<ParentchainBlock, _, _, _, _>(
+		blocks,
+		opaque_calls,
+		propose_to_block_import_ocall_api,
+		&validator_access,
+		&extrinsics_factory,
+	)
+	.unwrap();
+
+	// After importing the sidechain block, the trusted operation should be removed.
+	assert!(rpc_author.get_pending_tops_separated(shard_id).unwrap().0.is_empty());
+
+	// After importing the block, the state hash must be changed.
+	assert_ne!(
+		state_hash_before_block_production,
+		get_state_hash(state_handler.as_ref(), &shard_id)
+	);
```

Comment:
We could even calculate the expected post-import statehash.

Reply:
Yes we could - but it would add quite a bit of code to do so, no? And making sure that the state hashes are correct, is part of the unit tests? So I don't think we have to test every detail here, we just want to ensure the operation works from a high-level perspective.

Reply:
Ok, I understand your argument. You are right about that. However, I think it wouldn't be a big effort. We only have the to decrypt the state_update from the block and apply to the state before we imported the block.